code/Model/VirtualPage.php

```diff
@@ -368,6 +369,22 @@ public function CMSTreeClasses()
         return parent::CMSTreeClasses() . ' VirtualPage-' . $this->CopyContentFrom()->ClassName;
     }
 
+    /**
+     * Use the target page's class name for fetching templates - as we need to take on its appearance
+     *
+     * @param string $action
+     * @return array
+     */
+    public function getViewerTemplates($action = '')
+    {
+        $copy = $this->CopyContentFrom();
+        if ($copy && $copy->exists()) {
+            return SSViewer::get_templates_by_class($copy->ClassName, $action, SiteTree::class);
```

Comment:
would `get_class($copy)` work for this case?

oh, possibly because of VirtualPage inception?

Reply:
No I think that should work actually, I’ll give it a shot tomorrow morning unless someone wants to push to my fork in the meantime

Reply:
Looks good overall, maybe just make that change and we'll approve.